packages/teleterm/src/services/config/configService.ts

```diff
@@ -1,39 +1,180 @@
-import { merge } from 'lodash';
+import { z } from 'zod';
 
-import { keyboardShortcutsConfigProvider } from './providers/keyboardShortcutsConfigProvider';
-import { appearanceConfigProvider } from './providers/appearanceConfigProvider';
-import { Config, ConfigService, ConfigServiceProvider } from './types';
+import { FileStorage } from 'teleterm/services/fileStorage';
+import { Platform } from 'teleterm/mainProcess/types';
 
-type ConfigServiceProviders = {
-  readonly [Property in keyof Config]: ConfigServiceProvider<Config[Property]>;
-};
+import { createConfigStore } from './configStore';
 
-export class ConfigServiceImpl implements ConfigService {
-  private config: Config;
-  private configProviders: ConfigServiceProviders = {
-    keyboardShortcuts: keyboardShortcutsConfigProvider,
-    appearance: appearanceConfigProvider,
-  };
+const appConfigSchema = z.object({
+  'keymap.tab-1': z.string(),
+  'keymap.tab-2': z.string(),
+  'keymap.tab-3': z.string(),
+  'keymap.tab-4': z.string(),
+  'keymap.tab-5': z.string(),
+  'keymap.tab-6': z.string(),
+  'keymap.tab-7': z.string(),
+  'keymap.tab-8': z.string(),
+  'keymap.tab-9': z.string(),
+  'keymap.tab-close': z.string(),
+  'keymap.tab-new': z.string(),
+  'keymap.tab-previous': z.string(),
+  'keymap.tab-next': z.string(),
+  'keymap.toggle-connections': z.string(),
+  'keymap.toggle-clusters': z.string(),
+  'keymap.toggle-identity': z.string(),
+  'keymap.open-quick-input': z.string(),
```

Comment:
I'm wondering about converting them to camelCase (`keymap.tab1`) so they will match the last 3 keys.

Reply:
Done, although we should also rename the keyboard shortcuts that are used in the app, like `tab-1` (not config key `keymap.tab1`), but it will affect many files, so I would do it in another PR.